src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs

```diff
@@ -137,8 +183,90 @@ public static string GetDefaultFixAllTitle(FixAllContext fixAllContext)
                 FixAllScope.Document => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_1, diagnosticId, triggerDocument!.Name),
                 FixAllScope.Project => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_1, diagnosticId, triggerProject.Name),
                 FixAllScope.Solution => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Solution, diagnosticId),
+                FixAllScope.ContainingMember => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Containing_Member, diagnosticId),
+                FixAllScope.ContainingType => string.Format(WorkspaceExtensionsResources.Fix_all_0_in_Containing_Type, diagnosticId),
                 _ => throw ExceptionUtilities.UnexpectedValue(fixAllScope),
             };
         }
+
+        public static async Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetDocumentsAndSpansForContainingMemberOrTypeAsync(
+            Document document, FixAllScope fixAllScope, TextSpan triggerSpan, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var decl = await GetContainingMemberOrTypeDeclarationAsync(document, fixAllScope, triggerSpan, cancellationToken).ConfigureAwait(false);
+            if (decl == null)
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
+
+            if (fixAllScope == FixAllScope.ContainingMember)
+            {
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                    .SetItem(document, ImmutableArray.Create(decl.FullSpan));
+            }
+            else
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
this seems to be C#/VB specific, whcih concerns me.

Reply:
Yes, that is intentionally so for this PR to keep it simple and limit the newly added scopes to just C#/VB for now. As mentioned in https://github.com/dotnet/roslyn/pull/59989#discussion_r821310498, we can absolutely enhance this in future to enable TS/F# to support these scopes, for which we would need to extract out all the code to map a trigger span to containing member/type span to some language service. I didn't want to add it here to avoid making this PR unnecessary bulkier.